### PR TITLE
FeatureToggles: Allow changing prod env safe feature toggles via URL

### PR DESCRIPTION
**Pull request by adela-almasan**
This PR allows setting the feature flags via URL for prod environments, if the flag is considered safe. Current safe list includes deprecation related feature flags.

```['autoMigrateOldPanels', 'autoMigrateGraphPanel', 'autoMigrateTablePanel', 'autoMigratePiechartPanel', 'autoMigrateWorldmapPanel', 'autoMigrateStatPanel, disableAngular']``` 

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by ryantxu — APPROVED** [review]
LGTM -- testing many of the forced auto-migration we will do for 11 is way to difficult without this

File: packages/grafana-runtime/src/config.ts [hunk 1]
```diff
@@ -238,16 +235,32 @@ function overrideFeatureTogglesFromLocalStorage(config: GrafanaBootConfig) {
...
 
+  const isLocalDevEnv = config.buildInfo.env === 'development';
+
+  const prodUrlAllowedFeatureFlags = new Set([
+    'autoMigrateOldPanels',
+    'autoMigrateGraphPanel',
+    'autoMigrateTablePanel',
+    'autoMigratePiechartPanel',
+    'autoMigrateWorldmapPanel',
+    'autoMigrateStatPanel, disableAngular',
```
Comment (xnyo) [inline packages/grafana-runtime/src/config.ts:251]: Maybe I am missing something but shouldn't this be:

```suggestion
    'autoMigrateStatPanel',
    'disableAngular',
```

Comment (adela-almasan) [reply packages/grafana-runtime/src/config.ts:251]: good catch, yes it should!

File: packages/grafana-runtime/src/config.ts [hunk 2]
```diff
@@ -238,16 +235,33 @@ function overrideFeatureTogglesFromLocalStorage(config: GrafanaBootConfig) {
   }
 }
 
-function overrideFeatureTogglesFromUrl(config: GrafanaBootConfig) {
+function overrideFeatureTogglesFromUrl(config: GrafanaBootConfig, allowWhiteListed = false) {
```
Comment (nmarrs) [inline packages/grafana-runtime/src/config.ts:238]: Are we using the `allowWhiteListed` parameter at all?

```suggestion
function overrideFeatureTogglesFromUrl(config: GrafanaBootConfig) {
```

**Review by nmarrs — APPROVED** [review]
minor nit but looks good!

File: packages/grafana-runtime/src/config.ts (hunk 2 above)
Comment (adela-almasan) [reply packages/grafana-runtime/src/config.ts:238]: oops, no. leftover

**Review by xnyo — APPROVED** [review]
LGTM! Awesome :rocket:

**Pull request merged**